### PR TITLE
Added content describing difference between session and contract code

### PR DESCRIPTION
**Pull request by GloreenS**
### Related links

Card #609

### CODE REVIEW

**Review by ACStoneCL — APPROVED** [review]
I think the content here is good, my only suggestion would be to place it between "Unit Testing Session Code" and "Writing a Basic Smart Contract in Rust" in the sidebar.

I think using it as a delineator between the Session Code docs and the Contract Code docs would help get the idea across.

**Comment by ipopescu** [comment]
@Shalmali123 @ramujin please take a look as well.

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# Difference Between Session Code and Contract Code
+
+| Session Code | Contract Code |
+| --- | --- |
+| Session code always executes in the context of the account that signed the deploy that contains the session code. | Contract code executes within the context of its own state. |
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md:5]: @GloreenS @ACStoneCL can we add a bit more detail here explaining what a contract's state is? Maybe something like this:
```suggestion
| Session code always executes in the context of the account that signed the deploy that contains the session code. | Contract code executes within the context of its own state, meaning the contract's abstract environment containing [named keys](/dapp-dev-guide/sdkspec/types_chain/#namedkey). |
```

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+| When a `put_key` call is made within the body of the session code, the key is added to the account's named keys.| When `put_key` call is made within the contract's execution, the key is inserted into the contract's context. So, the key will appear in the contract's named keys.|
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md:6]: Small typo
```suggestion
| When a `put_key` call is made within the body of the session code, the key is added to the account's named keys.| When a `put_key` call is made within the contract's execution, the key is inserted into the contract's context. So, the key will appear in the contract's named keys.|
```

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+| Session code has only one entry point, that is the `call` function, which you can use to interact with the session code.| A contract can have multiple entry points that will help you interact with the contract code.|  
+| Any action taken by the session code is initiated by the `call` function within the session code. | Any action undertaken by a contract must initiate through an outside call, usually via session code.|
+
+## Sample Session Code
+
+This session code issues a delegation request to a pre-installed system contract. It accepts a delegator's public key, validator's public key, amount and a delegation rate and issues a delegation request to the auction contract.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md:12]:
```suggestion
This session code issues a delegation request to a pre-installed system contract. It accepts a delegator's public key, a validator's public key, the amount to delegate, and a delegation rate. It then issues a delegation request to the auction contract. 
```

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    let amount = runtime::get_named_arg(ARG_AMOUNT);
+
+    delegate(delegator, validator, amount);
+}
+
+```
+
+## Sample Contract Code
+
+This contract code helps with key management. The various entry points such as `set_key_weight`, `set_deployment_threshold`, and `set_key_management_threshold` allow  you to interact with the contract code and manage the contract's keys.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md:56]: @darthsiroftardis, I thought we said that we would not include key management contract code in the documentation until the code gets revisited. 
@GloreenS @darthsiroftardis please revisit this code and perhaps include another example contract, such as a snippet from the NFT contract.

**Review by ipopescu — CHANGES_REQUESTED** [review]
@GloreenS @darthsiroftardis @ACStoneCL I am requesting changes because I am concerned about the use of the key management contract here. Please review, and perhaps a different contract can be used until the key management contract can be published again. 
Second, there is a duplicate section in the docs that needs to be updated.

**Review by darthsiroftardis — CHANGES_REQUESTED** [review]
Lets remove all code samples from this page and just focus on the informational content

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md]: I see Gloreen applied the change. Resolving the conversation.

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md (hunk 4 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/writing-contracts/session-contract-diff.md]: Code was removed.

**Comment by ipopescu** [comment]
> @GloreenS are you planning to keep this section as well: https://docs.casperlabs.io/dapp-dev-guide/writing-contracts/session-code/#comparing-session-code-and-contract-code. I recommend removing it as it is a duplicate, or you can link to the new page created.

This is the only comment that I have remaining. All others have been resolved. Thank you!

**Review by darthsiroftardis — APPROVED** [review]

File: source/docs/casper/dapp-dev-guide/writing-contracts/session-code.md [hunk 5]
```diff
@@ -14,13 +14,7 @@ Session code is the simplest piece of logic you can execute on a Casper Network.
...
 
-### Comparing Session Code and Contract Code
-The following points try to explain the difference between session code and contract code:
-
-- Session code and contract code run in two different type of contexts. Session code always executes in the context of the account that signed the deploy that contains the session code. This means that when a `put_key` call is made within the body of the session code, the key is added to the account's named keys. 
-- Conversely, contract code executes in its own context. Which means that when `put_key` call is made within the contract's execution, the key is inserted into the contract's context. So, the key will appear in the contract's named keys.
-- Session code has only one entry point, that is the `call` function, which you can use to interact with the session code. 
-- A contract can have multiple entry points that will help you interact with the contract code. 
+To understand the difference between session code and contract code, see [Difference Between Session Code and Smart Contract](../writing-contracts/rust.md/#difference-between-session-code-and-smart-contract)
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/session-code.md:17]:
```suggestion
To understand the difference between session code and contract code, see [Difference Between Session Code and Smart Contract](../writing-contracts/rust.md/#difference-between-session-code-and-smart-contract).
```

**Review by ipopescu — APPROVED** [review]

**Pull request merged**